provider/equinix/environ.go

```diff
@@ -188,25 +187,34 @@ func (e *environ) InstanceTypes(context.ProviderCallContext, constraints.Value)
 }
 
 func (e *environ) Instances(ctx context.ProviderCallContext, ids []instance.Id) ([]instances.Instance, error) {
-	toReturn := []instances.Instance{}
+	toReturn := make([]instances.Instance, len(ids))
+	var missingInstanceCount int
 
 	tags := set.NewStrings("juju-model-uuid=" + e.Config().UUID())
 
-	for _, id := range ids {
+	for i, id := range ids {
 		//TODO handle case when some of the instanes are missing
-		d, _, err := e.equnixClient.Devices.Get(string(id), nil)
-		if err != nil {
+		d, resp, err := e.equnixClient.Devices.Get(string(id), nil)
+		if err != nil && resp.Request.Response.StatusCode == 404 {
+			logger.Warningf("instance %s not found", string(id))
+			missingInstanceCount = missingInstanceCount + 1
+		} else if err != nil {
 			return nil, errors.Annotatef(err, "looking up device with ID %q", id)
 		}
+
 		deviceTags := set.NewStrings(d.Tags...)
 		if !tags.Intersection(deviceTags).IsEmpty() {
-			toReturn = append(toReturn, &equnixDevice{e, d})
+			toReturn[i] = &equnixDevice{e, d}
 		}
 	}
-	if len(toReturn) == 0 {
-		return nil, environs.ErrNoInstances
+
+	if missingInstanceCount > 0 {
+		if missingInstanceCount == len(toReturn) {
+			return nil, environs.ErrNoInstances
+		}
```

Comment:
This is wrong. If you don't return `ErrNoInstances` you should return `return toReturn, environs.ErrPartialInstances` here

Reply:
Oh, nevermind; this got fixed in the followup commit...